packages/viz/src/utils/basal.js

```diff
@@ -29,11 +29,13 @@ import { ONE_HR } from "./datetime";
 */
 export function getBasalSequences(basals) {
   const basalSequences = [];
+  if (!Array.isArray(basals) || basals.length < 1) {
+    return basalSequences;
+  }
   let currentBasal = basals[0];
   let seq = [basals[0]];
 
-  let idx = 1;
-  while (idx <= basals.length - 1) {
+  for (let idx = 1; idx < basals.length; idx++) {
```

Comment:
Is there a reason for changing `idx <= basals.length` to `idx < basals.length` :question:

Reply:
To use a standard for loop, instead of while() for a simple counter

Reply:
What I mean is that now, we don't execute the code in the loop if `idx == basals.length`, is it wanted?

Reply:
Yes, below the loop, the leftover basal is added, no change here.
It was already the case previously.